### ISSUE TITLE
Recursive crash

### ISSUE THREAD

**Issue by iwasrobbed** [calimarkus/JDStatusBarNotification#24]
It appears there is a scenario where infinite recursion can occur when the underlying view controller is `JDStatusBarNotificationViewController`:

https://github.com/jaydee3/JDStatusBarNotification/blob/master/JDStatusBarNotification/JDStatusBarNotification.m#L506

``` objC
- (BOOL)shouldAutorotate {
    return [[self keyWindowRootViewController] shouldAutorotate];
}
```

Full crash log for you: https://gist.github.com/iwasrobbed/9d1cf7ddc96413023d7c

**Comment by waltsellers** [comment]
For us, it occurred if it overlaps with the presentation of a UIAlertView.

**Comment by victorengel** [comment]
Still seeing this issue -- anyone else? I found this issue because there is a comment in the code:
    // find another window, if our window is the key window (should fix #24)

So I'm using the "fixed" version.
